build-support/bin/mypy.py

```diff
@@ -6,7 +6,7 @@
 
 
 def main() -> None:
-  subprocess.run(["./pants", "--tag=+type_checked", "mypy", "::"], check=True)
+  subprocess.run(["./pants", "--tag=+type_checked", "--lint-skip", "--no-lint-mypy-skip", "lint", "::"], check=True)
```

Comment:
I think you can instead set this to `lint.mypy` and ignore the `--lint-skip` etc. Also, with your new whitelist feature, the `--tag=+type_checked` should be removed.

Reply:
turns out lint.mypy isn't enough to just run mypy, one must specify to skip all lint tasks except mypy

Reply:
Yep, @mabdi3 is right: all tasks in a goal run if any do.